etna/lib/etna/clients/metis/models.rb

```diff
@@ -281,6 +281,13 @@ def initialize(raw = {})
           @raw = raw
         end
 
+        def with_containing_folder(folder)
```

Comment:
This simply allows translating from the "hide_paths: true" to full path folder objects.
It is amusing, but full path resolution of a file is an expensive operation just due to the potential recursive aspects that make single scan lookups hard (won't know what precise order you will find parent folders in before you look them up).  

Incidentally, the attribute processors will no use watch folders and not use hide_paths: true, and I believe this further increases performance.